### PR TITLE
Added tab cycling to Completer

### PR DESCRIPTION
**Pull request by cameron-toy**
cycle2

Added tab and shift + tab cycling to auto completion.

## References

[Cycle completion with tab and automatically insert parenthesis #9082](https://github.com/jupyterlab/jupyterlab/issues/9082)
[Autocomplete cycle options using <KEY> #5863](https://github.com/jupyterlab/jupyterlab/issues/5863)

## User-facing changes
Cycling through auto-complete entries with the arrow keys or the tab key wraps bottom->top and top->bottom. Not sure if this should be the behavior for arrow keys and tab cycling, just tab cycling, or neither.

Note: I had a one-time issue with the shift-tab key combination opening a tooltip in the console but haven't been able to replicate it.

### CODE REVIEW

**Comment by blink1073** [comment]
cc @afshin @krassowski

File: packages/completer/src/widget.ts [hunk 1]
```diff
@@ -439,6 +439,8 @@ export class Completer extends Widget {
         if (populated) {
           this.update();
         }
+
+        this._cycle(event.shiftKey ? 'up' : 'down');
```
Comment (krassowski) [inline packages/completer/src/widget.ts:443]: I am not sure if having discordant keybindings is a good thing, this is:
- <kbd>shift</kbd> + <kbd>Tab</kbd> means "go up", but
- <kbd>Tab</kbd> meaning "insert active completion item"
might end up confusing users and lead to frustration ("it works one way, why doesn't it work the other way too!").

Would it be possibly better to keep only arrows for cycling for this PR, and in another PR add a capability to change the keybindings from "arrows for cycling and Tab for insert" to "Tab/Shift Tab for cycling and Enter for insert"?

Comment (krassowski) [reply packages/completer/src/widget.ts:443]: Nevermind I misunderstood it.

**Review by krassowski — APPROVED** [review]
Looks great to me.

**Review by krassowski — CHANGES_REQUESTED** [review]
While the UX and code looks great, the tests need to be updated too. In particular these tests of completer/widget (in test/widget.spec.ts) are failing because the behaviour changed:

```
✕ should select the item below and not progress past last
✕ should select the completion item below and not progress past last
✕ should select the item above and not progress beyond first
✕ should select the completion item above and not progress beyond first
```

**Comment by cameron-toy** [comment]
> While the UX and code looks great, the tests need to be updated too. In particular these tests of completer/widget (in test/widget.spec.ts) are failing because the behaviour changed:
> 
> ```
> ✕ should select the item below and not progress past last
> ✕ should select the completion item below and not progress past last
> ✕ should select the item above and not progress beyond first
> ✕ should select the completion item above and not progress beyond first
> ```

Okay, updated tests. I also added tests specifically for tab cycling.

Also not too sure about the names.

**Review by krassowski — APPROVED** [review]

**Comment by krassowski** [comment]
> Also not too sure about the names.

"wrap to" sounds fine as a reference to [wraparound](https://en.wikipedia.org/wiki/Wraparound_(video_games)); alternatively "cycle to" would be fine too (for consistency with the documentation of `_cycle` function).

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**